source/nvwave.py

```diff
@@ -362,11 +385,14 @@ def _feedUnbuffered(self, data, onDone=None):
 		whdr.dwBufferLength = len(data)
 		with self._lock:
 			with self._waveout_lock:
-				self.open()  # required of close on idle see _idleUnbuffered
-				with self._global_waveout_lock:
-					winmm.waveOutPrepareHeader(self._waveout, LPWAVEHDR(whdr), sizeof(WAVEHDR))
-					winmm.waveOutWrite(self._waveout, LPWAVEHDR(whdr), sizeof(WAVEHDR))
-			self.sync()
+				self.open()  # required if close on idle see _idleUnbuffered
+				if self._prevOnDone is not self.STOPPING:
+					# If we are stopping, waveOutReset has already been called.
+					# Pushing more data confuses the state of nvWave
+					with self._global_waveout_lock:
+						winmm.waveOutPrepareHeader(self._waveout, LPWAVEHDR(whdr), sizeof(WAVEHDR))
+						winmm.waveOutWrite(self._waveout, LPWAVEHDR(whdr), sizeof(WAVEHDR))
+			self.sync()  # sync must still be called even if stopping, so that waveOutUnprepareHeader can be called
 			self._prev_whdr = whdr

```

Comment:
I think this is a bug, in the case that `nvWave` is stopping, we shouldn't save this unprepared header.

Reply:
But what then would unprepare that header? saving it off means that feed / sync will unprepare it next time (I think) not saving it, and setting _prev_whdr to None, you'd have to specifically call waveOutUnprepare for hdr I think?